File: tags/2.1.0/osznconnection-file-handling/src/main/java/org/complitex/osznconnection/file/web/component/DataRowHoverBehavior.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package org.complitex.osznconnection.file.web.component;

import org.apache.wicket.Component;
import org.apache.wicket.ajax.AjaxRequestTarget;
import org.apache.wicket.behavior.Behavior;
import org.apache.wicket.markup.html.IHeaderResponse;
import org.apache.wicket.request.resource.PackageResourceReference;

/**
 *
 * @author Artem
 */
public final class DataRowHoverBehavior extends Behavior {

    public DataRowHoverBehavior() {
    }

    @Override
    public void renderHead(Component component, IHeaderResponse response) {
        response.renderCSSReference(new PackageResourceReference(DataRowHoverBehavior.class,
                DataRowHoverBehavior.class.getSimpleName() + ".css"));
        response.renderJavaScriptReference(new PackageResourceReference(DataRowHoverBehavior.class,
                DataRowHoverBehavior.class.getSimpleName() + ".js"));
    }

    public void deactivateDataRow(AjaxRequestTarget target) {
        target.appendJavaScript("(function(){ $('table tr.data-row.data-row-hover').removeClass('data-row-hover'); })();");
    }
}

File: osznconnection-file-handling/src/main/java/org/complitex/osznconnection/file/web/component/DataRowHoverBehavior.js
$(function(){
    (function(){
        $("table tr.data-row").on('click', 'a.data-row-link',function(){
            var link = $(this);
            var row = link.closest("tr.data-row");
            row.addClass("data-row-hover");
        });

        $("table tr.data-row-link").bind("click", function(){
            var row = $(this);
            row.addClass("data-row-hover");
        });
    })();
});